refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java

```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
+                UndoableWrapper undoableWrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
+                commits.forEach(commit -> {
+                    if (undoableWrapper != null) {
+                        undoableWrapper.setActive(true, this);
                     }
-
-                    if(commit instanceof ProgressProvider) {
+                    if (commit instanceof ProgressProvider) {
                         ProgressProvider progressProvider = (ProgressProvider) commit;
                         progressProvider.addProgressListener(progressListener);
-                    }
-                    try {
-                        commit.commit();
-                    } finally {
-                        if(commit instanceof ProgressProvider) {
-                            ProgressProvider progressProvider = (ProgressProvider) commit;
+                        try {
+                            commit.commit();
+                        } finally {
                             progressProvider.removeProgressListener(progressListener);
                         }
+                    } else {
+                        commit.commit();
                     }
-                    if (wrapper != null) {
-                        unsetWrappers(commit, wrapper);
+                    if (undoableWrapper != null) {
+                        undoableWrapper.setActive(false, null);
                     }
+                });
+                if (undoableWrapper != null) {
+                    undoableWrapper.close();
                 }
-                if (wrapper != null) {
-                    wrapper.close();
-                }
-                for (Transaction commit : commits) {
-                    SPIAccessor.DEFAULT.sum(commit);
-                }
+                commits.forEach(commit -> SPIAccessor.DEFAULT.sum(commit));
             }
             if (saveAfterDone) {
                 LifecycleManager.getDefault().saveAll();
-                for (DataObject dob:DataObject.getRegistry().getModified()) {
-                    SaveCookie cookie = dob.getCookie(SaveCookie.class);
+                for (DataObject dataObject : DataObject.getRegistry().getModified()) {
+                    SaveCookie cookie = dataObject.getLookup().lookup(SaveCookie.class);
                     try {
                         cookie.save();
                     } catch (IOException ex) {
                         Exceptions.printStackTrace(ex);
                     }
                 }
             }
-            for (RefactoringElementImplementation fileChange:SPIAccessor.DEFAULT.getFileChanges(bag)) {
-                if (fileChange.isEnabled()) {
-                    fileChange.performChange();
-                }
-            }
+            SPIAccessor.DEFAULT.getFileChanges(bag).stream().filter(fileChange -> fileChange.isEnabled()).forEachOrdered(fileChange -> fileChange.performChange());
```

Comment:
Well, I sort of doubt that this stream syntax improves readability over (more verbose) `for` or `forEach`. But please make a breakline between the individual processing steps, like this:
```
SPIAccessor.DEFAULT.getFileChanges(bag).stream().
    filter(fileChange -> fileChange.isEnabled()).
    forEachOrdered(fileChange -> fileChange.performChange());
```

Which can be usually scanned faster. Debugging of streamized code is a real pain, BTW.

Reply:
You are right. Would prefer, if the dot is placed at the new line, in front of the method call.

Reply:
Right. I think that placing the dot first and indenting twice is more standard:

```java
SPIAccessor.DEFAULT.getFileChanges(bag).stream()
        .filter(fileChange -> fileChange.isEnabled())
        .forEachOrdered(fileChange -> fileChange.performChange());
```